<Type Name="SemaphoreFullException" FullName="System.Threading.SemaphoreFullException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="465d12c63165fb84e4220d19a875fd17949ddf2c" />
    <Meta Name="ms.sourcegitcommit" Value="6a0b904069161bbaec4ffd02aa7d9cf38c61e72e" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="es-ES" />
    <Meta Name="ms.lasthandoff" Value="06/24/2018" />
    <Meta Name="ms.locfileid" Value="36605979" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class SemaphoreFullException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit SemaphoreFullException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.Threading.SemaphoreFullException" />
  <TypeSignature Language="VB.NET" Value="Public Class SemaphoreFullException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class SemaphoreFullException : SystemException" />
  <TypeSignature Language="F#" Value="type SemaphoreFullException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>System.Threading</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=2.0.0.0, Culture=Neutral, PublicKeyToken=b77a5c561934e089")</AttributeName>
    </Attribute>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(false)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="31c9a-101">Excepción que se produce cuando se llama al método <see cref="Overload:System.Threading.Semaphore.Release" /> en un semáforo cuyo recuento ya ha alcanzado el valor máximo.</span>
      <span class="sxs-lookup">
        <span data-stu-id="31c9a-101">The exception that is thrown when the <see cref="Overload:System.Threading.Semaphore.Release" /> method is called on a semaphore whose count is already at the maximum.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="31c9a-102">El recuento en un semáforo es disminuye cada vez que un subproceso entra en el semáforo y aumenta cuando un subproceso libere el semáforo.</span><span class="sxs-lookup"><span data-stu-id="31c9a-102">The count on a semaphore is decremented each time a thread enters the semaphore, and incremented when a thread releases the semaphore.</span></span> <span data-ttu-id="31c9a-103">Cuando el recuento es cero, las solicitudes posteriores se bloquean hasta que otros subprocesos liberan el semáforo.</span><span class="sxs-lookup"><span data-stu-id="31c9a-103">When the count is zero, subsequent requests block until other threads release the semaphore.</span></span> <span data-ttu-id="31c9a-104">Cuando todos los subprocesos han liberado el semáforo, el recuento es el valor máximo especificado cuando se creó el semáforo.</span><span class="sxs-lookup"><span data-stu-id="31c9a-104">When all threads have released the semaphore, the count is at the maximum value specified when the semaphore was created.</span></span> <span data-ttu-id="31c9a-105">Si un error de programación hace que un subproceso llamar a la <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> método en este momento, un <xref:System.Threading.SemaphoreFullException> se produce.</span><span class="sxs-lookup"><span data-stu-id="31c9a-105">If a programming error causes a thread to call the <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> method at this point, a <xref:System.Threading.SemaphoreFullException> is thrown.</span></span>  
  
> [!NOTE]
>  <span data-ttu-id="31c9a-106">El <xref:System.Threading.Semaphore> clase no exige la identidad del subproceso en las llamadas a la <xref:System.Threading.WaitHandle.WaitOne%2A?displayProperty=nameWithType> y <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> métodos.</span><span class="sxs-lookup"><span data-stu-id="31c9a-106">The <xref:System.Threading.Semaphore> class does not enforce thread identity on calls to the <xref:System.Threading.WaitHandle.WaitOne%2A?displayProperty=nameWithType> and <xref:System.Threading.Semaphore.Release%2A?displayProperty=nameWithType> methods.</span></span> <span data-ttu-id="31c9a-107">No es necesario para el mismo subproceso que llamó a <xref:System.Threading.WaitHandle.WaitOne%2A> para llamar a <xref:System.Threading.Semaphore.Release%2A>.</span><span class="sxs-lookup"><span data-stu-id="31c9a-107">It is not necessary for the same thread that called <xref:System.Threading.WaitHandle.WaitOne%2A> to call <xref:System.Threading.Semaphore.Release%2A>.</span></span>  
  
 <span data-ttu-id="31c9a-108"><xref:System.Threading.SemaphoreFullException> no indica necesariamente un problema con el código donde se produjo la excepción.</span><span class="sxs-lookup"><span data-stu-id="31c9a-108"><xref:System.Threading.SemaphoreFullException> does not necessarily indicate a problem with the code where the exception occurred.</span></span> <span data-ttu-id="31c9a-109">Considere el siguiente escenario: subprocesos A y B entran en un semáforo que tiene un recuento máximo de dos.</span><span class="sxs-lookup"><span data-stu-id="31c9a-109">Consider the following scenario: Thread A and thread B enter a semaphore that has a maximum count of two.</span></span> <span data-ttu-id="31c9a-110">Un error de programación en el subproceso B hace que llame a <xref:System.Threading.Semaphore.Release%2A> dos veces, para que el recuento del semáforo está lleno.</span><span class="sxs-lookup"><span data-stu-id="31c9a-110">A programming error in thread B causes it to call <xref:System.Threading.Semaphore.Release%2A> twice, so that the count on the semaphore is full.</span></span> <span data-ttu-id="31c9a-111">Como resultado, cuando el subproceso A finalmente llama <xref:System.Threading.Semaphore.Release%2A>, un <xref:System.Threading.SemaphoreFullException> se produce.</span><span class="sxs-lookup"><span data-stu-id="31c9a-111">As a result, when thread A eventually calls <xref:System.Threading.Semaphore.Release%2A>, a <xref:System.Threading.SemaphoreFullException> is thrown.</span></span>  
  
 <span data-ttu-id="31c9a-112">Para obtener una lista con los valores de propiedad iniciales de una instancia de la clase <xref:System.Threading.SemaphoreFullException>, vea el constructor <xref:System.Threading.SemaphoreFullException.%23ctor>.</span><span class="sxs-lookup"><span data-stu-id="31c9a-112">For a list of initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class, see the <xref:System.Threading.SemaphoreFullException.%23ctor> constructor.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="31c9a-113">En el ejemplo de código siguiente se muestra cómo puede producir un error de programación en un subproceso un <xref:System.Threading.SemaphoreFullException> en otro subproceso: dos subprocesos entran en un semáforo.</span><span class="sxs-lookup"><span data-stu-id="31c9a-113">The following code example shows how a programming error in one thread can lead to a <xref:System.Threading.SemaphoreFullException> in another thread: Two threads enter a semaphore.</span></span> <span data-ttu-id="31c9a-114">El segundo subproceso libera el semáforo dos veces, mientras que el primer subproceso todavía se está ejecutando la tarea.</span><span class="sxs-lookup"><span data-stu-id="31c9a-114">The second thread releases the semaphore twice, while the first thread is still executing its task.</span></span> <span data-ttu-id="31c9a-115">Cuando el primer subproceso finalice y libere el semáforo, el recuento del semáforo ya está completo y se produce una excepción.</span><span class="sxs-lookup"><span data-stu-id="31c9a-115">When the first thread finishes and releases the semaphore, the semaphore count is already full and an exception is thrown.</span></span>  
  
 [!code-cpp[System.Threading.SemaphoreFullException#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/CPP/source.cpp#1)]
 [!code-csharp[System.Threading.SemaphoreFullException#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/CS/source.cs#1)]
 [!code-vb[System.Threading.SemaphoreFullException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Threading.SemaphoreFullException/VB/source.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Threading.Semaphore" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="31c9a-116">Inicializa una nueva instancia de la clase <see cref="T:System.Threading.SemaphoreFullException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-116">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="31c9a-117">Inicializa una nueva instancia de la clase <see cref="T:System.Threading.SemaphoreFullException" /> con valores predeterminados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-117">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with default values.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="31c9a-118">Este constructor inicializa la <xref:System.Exception.Message%2A> propiedad de la nueva instancia en un mensaje proporcionado por el sistema que describe el error, como "Agregar un recuento dado al semáforo hará que se supere el recuento máximo."</span><span class="sxs-lookup"><span data-stu-id="31c9a-118">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Adding the given count to the semaphore would cause it to exceed its maximum count."</span></span> <span data-ttu-id="31c9a-119">Este mensaje tiene en cuenta la referencia cultural del sistema actual.</span><span class="sxs-lookup"><span data-stu-id="31c9a-119">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="31c9a-120">En la siguiente tabla se muestran los valores iniciales de las propiedades de una instancia de la clase <xref:System.Threading.SemaphoreFullException>.</span><span class="sxs-lookup"><span data-stu-id="31c9a-120">The following table shows the initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class.</span></span>  
  
|<span data-ttu-id="31c9a-121">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="31c9a-121">Property</span></span>|<span data-ttu-id="31c9a-122">Valor</span><span class="sxs-lookup"><span data-stu-id="31c9a-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="31c9a-123">`null`.</span><span class="sxs-lookup"><span data-stu-id="31c9a-123">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="31c9a-124">La cadena de mensaje de error proporcionado por el sistema.</span><span class="sxs-lookup"><span data-stu-id="31c9a-124">The system-supplied error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : string -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="31c9a-125">Mensaje de error que explica el motivo de la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-125">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="31c9a-126">Inicializa una nueva instancia de la clase <see cref="T:System.Threading.SemaphoreFullException" /> con el mensaje de error especificado.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-126">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="31c9a-127">El contenido de `message` es una cadena de texto diseñada para informar al usuario sobre la excepción.</span><span class="sxs-lookup"><span data-stu-id="31c9a-127">The content of `message` is a text string intended to inform the user about the exception.</span></span> <span data-ttu-id="31c9a-128">El autor de llamada de este constructor debe asegurarse de que la cadena se haya adaptado para la referencia cultural actual del sistema.</span><span class="sxs-lookup"><span data-stu-id="31c9a-128">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="31c9a-129">La tabla siguiente muestran los valores de propiedad iniciales para una instancia de la <xref:System.Threading.SemaphoreFullException> clase inicializada con este constructor.</span><span class="sxs-lookup"><span data-stu-id="31c9a-129">The following table shows the initial property values for an instance of the <xref:System.Threading.SemaphoreFullException> class initialized with this constructor.</span></span>  
  
|<span data-ttu-id="31c9a-130">Propiedad.</span><span class="sxs-lookup"><span data-stu-id="31c9a-130">Property</span></span>|<span data-ttu-id="31c9a-131">Valor</span><span class="sxs-lookup"><span data-stu-id="31c9a-131">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="31c9a-132">`null`.</span><span class="sxs-lookup"><span data-stu-id="31c9a-132">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="31c9a-133">`message`.</span><span class="sxs-lookup"><span data-stu-id="31c9a-133">`message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected SemaphoreFullException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; SemaphoreFullException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="31c9a-134">Objeto <see cref="T:System.Runtime.Serialization.SerializationInfo" /> que contiene los datos serializados del objeto sobre la excepción iniciada.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-134">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> object that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="31c9a-135">Objeto <see cref="T:System.Runtime.Serialization.StreamingContext" /> que contiene información contextual sobre el origen o el destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-135">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> object that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="31c9a-136">Inicializa una nueva instancia de la clase <see cref="T:System.Threading.SemaphoreFullException" /> con datos serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-136">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with serialized data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="31c9a-137">Durante la deserialización se llama a este constructor para reconstituir el objeto de excepción transmitido en una secuencia.</span><span class="sxs-lookup"><span data-stu-id="31c9a-137">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="31c9a-138">Para obtener más información, consulte [serialización XML y SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="31c9a-138">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SemaphoreFullException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Threading.SemaphoreFullException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SemaphoreFullException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Threading.SemaphoreFullException : string * Exception -&gt; System.Threading.SemaphoreFullException" Usage="new System.Threading.SemaphoreFullException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Threading</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="31c9a-139">Mensaje de error que explica el motivo de la excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-139">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="31c9a-140">Excepción que es la causa de la excepción actual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-140">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="31c9a-141">Si el parámetro <c>innerException</c> no es <see langword="null" />, se genera la excepción actual en un bloque <see langword="catch" /> que controla la excepción interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-141">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="31c9a-142">Inicializa una nueva instancia de la clase <see cref="T:System.Threading.SemaphoreFullException" /> con el mensaje de error especificado y una referencia a la excepción interna que representa la causa de esta excepción.</span>
          <span class="sxs-lookup">
            <span data-stu-id="31c9a-142">Initializes a new instance of the <see cref="T:System.Threading.SemaphoreFullException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="31c9a-143">Una excepción que se produce como consecuencia directa de una excepción anterior debe incluir una referencia a esta última en la propiedad <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="31c9a-143">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="31c9a-144">La propiedad <xref:System.Exception.InnerException%2A> devuelve el mismo valor que se haya pasado al constructor, o bien `null` en caso de que la propiedad <xref:System.Exception.InnerException%2A> no le haya proporcionado al constructor el valor de la excepción interna.</span><span class="sxs-lookup"><span data-stu-id="31c9a-144">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Threading.Semaphore" />
      </Docs>
    </Member>
  </Members>
</Type>